---
title: "Import_QC_GSE52529-GPL16791"
output: html_document
---

Script and code follows "A step-by-step workflow for low-level analysis of single-cell RNA-seq data with Bioconductor". Aaron T. L. Lun1, Davis J. McCarthy2 and John C. Marioni. and the 

# Load and process data set: Trapnell2014 GSE52529-GPL16791
```{r load libraries, include=FALSE}
# load packages
pdf("results/QC_data/QC_plots_trapnell2014.pdf")
suppressPackageStartupMessages(library(MultiAssayExperiment))
suppressPackageStartupMessages(library(scater))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(plyr))
suppressPackageStartupMessages(library(scran))
suppressPackageStartupMessages(library(cowplot))

```

## load multiassay dataset from IMLS conquer

```{r load dataset}
# load multiassay dataset from IMLS conquer
maex <- readRDS("data/GSE52529-GPL16791.rds")
# summary of data set

print(maex)
dim(assay(maex)) #  65218  x 288


# extract the gene-level TPM and the counts obtaines by scaling the TPMs
cts <- assays(experiments(maex)[["gene"]])[["count_lstpm"]]
tpms <- assays(experiments(maex)[["gene"]])[["TPM"]]

```
# extract the Phenotype data
There are three different populations from different timepoints.
```{r phentotypes, echo=TRUE}
# extract the phenotype data

phn <- colData(maex)

# extract the cell annotation to phenoid column and rename it 
phn[, "phenoid"] <- as.character(interaction(as.data.frame(phn[, c("source_name_ch1")])))


phn[, "phenoid"] <- mapvalues(phn[, "phenoid"], from = c(grep("T0",phn[, "phenoid"], value=TRUE),grep("T24",phn[, "phenoid"], value=TRUE),grep("T48",phn[, "phenoid"], value=TRUE)) , to=c(array("T0", dim=96),array("T24", dim=96),array("T48", dim=96)) )


table(phn[, "phenoid"]) 

```
Have a closer look to the phenotype. Based on the platenames the experiment is confounded.
```{r have a closer look to pheno, echo=TRUE}
phn.tbl <- as_data_frame(phn)
glimpse(phn.tbl)
# extract the plate information
phn$title
grep("A01", phn$title, value=TRUE)
```

# create an SCEset
The expr slot is populated with the count data; obtained from the length-scalaed , count-scaled tpms. The values are automatically log2 transformed. gene-level tpms for the tpm slot.

```{r create SCEset, echo=TRUE }
# Create an SCEobject 
sceset <- newSCESet(countData = cts, 
                    phenoData = new("AnnotatedDataFrame", data = as.data.frame(phn)))
exprs(sceset)[1:3,1:3]

set_exprs(sceset, "tpm") <- tpms # tpm is populated with the gene level tpms

get_exprs(sceset, "tpm")[1:3,1:3] # untransformed tpm
get_exprs(sceset, "counts")[1:3,1:3] # untransformed counts
exprs(sceset)[1:3,1:3] # log2 transformed counts

```
# Identify the ERCCC spike-ins.
There are 92 ERCCC spike-inn genes. No mitochondrial genes , probably they are not annotaed yet.
```{r ERCC and mt genes }
is.spike <- grepl("^ERCC", rownames(sceset))
is.mito <- grepl("^mt-", rownames(sceset))
print( table(is.spike) )
print( table(is.mito) )

table(grepl("^ERCC", rownames(sceset))) 

hist(counts(sceset)["ERCC-00104",])

```
# reduce expression set
Keep only those features that are observed in at least 1 cell. 24106  features out of 65 k are filtered out. The original counts are used. Using this setting all but one ERCC spike inns are filtered out

```{r reduce}
keep_features <- rowSums(counts(sceset) > 0) >= 1
table(keep_features)
sceset <- sceset[keep_features, ]

```


# Compute some QC metrics
The ERCC is not set as a control feature. 3 cells were debris. Others cells were not well seperated, two ore more cells in a well.

```{r QC}


sceset <- calculateQCMetrics(sceset )
colnames(pData(sceset))


```

# Quality control on cells
Some cells have small library sizes (< 1 mio total counts) and show almost no expression. Candidates for filtering out.
```{r  histogramms }

size.med <- median(log10(sceset$total_counts), na.rm = TRUE)
size.mad <- mad(log10(sceset$total_counts), center = size.med, na.rm = TRUE)
lower.limit.size <- 10^(size.med - 3 * size.mad)/1e6
    
feature.med <- median(log10(sceset$total_features), na.rm = TRUE)
feature.mad <- mad(log10(sceset$total_features), center = size.med, na.rm = TRUE)
lower.limit.feat <- 10^(feature.med - 3 * feature.mad  )

pdf("results/QC_data/hist_Trapnell2014.pdf")

par(mfrow=c(1,2))
hist(sceset$total_counts/1e6, xlab="Library sizes ", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
abline(v=lower.limit.size,col=2, lty=2 )
hist(sceset$total_features, xlab="Number of expressed genes", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
abline(v=lower.limit.feat, col=2, lty=2 )

dev.off()

```
# Quality control using PCA and highest expressed genes
PCA plot based the quality metrics for each cell, e.g., the total number of reads, the total number of features and the proportion of mitochondrial or spike-in reads. 
```{r  Quality control using PCA }
p1 <- scater::plotPCA(sceset, exprs_values="exprs",pca_data_input="pdata",colour_by="phenoid")
p2 <- scater::plotQC(sceset, type = "find-pcs", variable = "total_features")


```

# Filter dataset gene wise and by librarysize 
Remove cells with log-library sizes that are more than 3 median absolute deviations (MADs) below the median log-library size. 52 cells were dropped according to this criteria, regarding the untransformed count data. add plot with threshold...
More than 10 % of the cells are removed, possibly there's a quality problem regarding the sequencing....
```{r save unfiltered data }

libsize.drop <- isOutlier(sceset$total_counts, nmads=3, type="lower", log=TRUE)
feature.drop <- isOutlier(sceset$total_features, nmads=3, type="lower", log=TRUE)
table(libsize.drop,feature.drop)

```
# Filter dataset by ERCCC spike inns
The amount of endogenous features should be constant. Cells with a high proportion should be removed. however as almost all cells have no spike-inns we dont use this filter. 
```{r  filter by spike inn }


```

Total 52  cells were filtered out. However there are still some wells with more than one cell and debris. For the the subsequent analysis these were filtered out as well. After filtering the cells by debris and by cell nummber 222 cells are remaing. One other criteria for filtering is if cells are controls.


```{r overview filtered cells }
is.debris <- pData(sceset)$characteristics_ch1.2=="debris: TRUE" # filter out destroyed cells
is.multiplecell <-  pData(sceset)$characteristics_ch1.4 != "cells in well: 1"
sceset.red <- sceset[,!(libsize.drop | feature.drop|is.debris| is.multiplecell )]

data.frame( ByLibSize=sum(libsize.drop), ByFeature=sum(feature.drop), ByCellnumber=sum(is.multiplecell), Bydebris=sum(is.debris )
    , Remaining=ncol(sceset.red) )


```
reduce expression matrix 

```{r reduce expression matrix, throwing away features that are not expressed. noninformative features}
keep_features <- rowSums(counts(sceset.red) > 0) >= 1e-04
table(keep_features)
sceset.red <- sceset.red[keep_features, ]
```

# Quality control using PCA and highest expressed genes
PCA plot based the quality metrics for each cell, e.g., the total number of reads, the total number of features and the proportion of mitochondrial or spike-in reads.  Clearly there are less extrem cells in the filterd dataset (possibly one or two).
Plotting the 50 highest expressed genes. No ERCC spike-inn in the top 50. The profile should look relatively "flat".
```{r  highest expressed genes}
p3 <- plotQC(sceset.red, type = "highest-expression", n=50)

```

```{r mean freq expression}
plotQC(sceset.red, type = "exprs-freq-vs-mean")
```




# Filtering the datasets based on the features. Here genes with low abundance are filtered out. We have to sets of features one is based on the mean expression and one on a expression in minium n cells. For the analysis the filter based on the mean expression is used.
```{r  Filtering out low-abundance genes }
ave.counts <- rowMeans(counts(sceset.red))
keep <- ave.counts >= 1
sum(keep)
ave.counts<- as.data.frame(log10(ave.counts))

p4 <- ggplot(ave.counts, aes(ave.counts))+geom_histogram(binwidth = 0.5)+geom_vline(aes(xintercept=log10(1), color=1))+guides(color = "none")+labs( x = expression(Log[10]~"average count"), y = "frequency")



sceset.red <- sceset.red[keep, ]


```

Normalize data according to Lun etal. 2016 using pooled counts to estimate size factors. The size factors in the non-filterd set contain non positive values what does this mean?
The size factor is correlated with the library sizes. As we have a homogenous cell population there's no obvious grouping. The difference of library size between cells is possibly due the seqeuncing depth.

```{r normalize through pool-based size factors }
sceset <- computeSumFactors(sceset, sizes=c(20, 40, 60, 80,120,240))
summary(sizeFactors(sceset))

sceset.red <- computeSumFactors(sceset.red, sizes=c(20, 40, 60, 80,120))
summary(sizeFactors(sceset.red))

plot(sizeFactors(sceset), sceset$total_counts/1e6, log="xy",
    ylab="Library size (millions)", xlab="Size factor")
plot(sizeFactors(sceset.red), sceset.red$total_counts/1e6, log="xy",
    ylab="Library size (millions)", xlab="Size factor")
```

The counts are normalized using log2 transformation and applying the size factors for normalized gene expression. Are really the size factors from Lun et al. used or the ones from edgeR?
```{r normalization }
sceset <- normaliseExprs(sceset)

sceset.red <- normaliseExprs(sceset.red)

```

# Save the normalized and cell filtered datasets
```{r save data }
res <- sceset
save(res,file = "data/sceset_org_GSE52529-GPL16791.rda")

res <- sceset.red
save(res,file = "data/sceset_red_GSE52529-GPL16791.rda")

```


# Identifying HVGs from the normalized log-expression

```{r High variable genes }

var.fit <- trendVar(sceset.red, trend="loess", use.spikes=FALSE, span=0.2) 
var.out <- decomposeVar(sceset.red, var.fit)


```


```{r High variable genes using the variance of the log-expression }

plot(var.out$mean, var.out$total, pch=16, cex=0.6, xlab="Mean log-expression",
    ylab="Variance of log-expression")
o <- order(var.out$mean)
lines(var.out$mean[o], var.out$tech[o], col="dodgerblue", lwd=2)



hvg.out <- var.out[which(var.out$FDR <= 0.05 & var.out$bio >= 0.5),]
hvg.out <- hvg.out[order(hvg.out$bio, decreasing=TRUE),]
nrow(hvg.out)
head(hvg.out)
plotExpression(sceset.red, rownames(hvg.out)[1:10])

```
# Select only the 4858 high variable genes and save the SCEset.
```{r}

sceset.hvg <-sceset.red[ rownames(hvg.out), ]
dim(sceset.hvg)

res <- sceset.hvg
save(res,file = "data/sceset_hvg_GSE52529-GPL16791.rda")

```


Plot the proportion of explained variances
```{r explained variance} 
expl_vars <- c( "phenoid","log10_total_counts", "log10_total_features", "pct_dropout",
               "pct_counts_top_200_features", "log10_counts_feature_controls",
               "pct_counts_feature_controls")
p5 <- plotQC(sceset.red, type = "explanatory-variables", variables = expl_vars)
```
plot tSNE represenatation
```{r tSNE}
p6 <- plotTSNE(sceset.red, exprs_values="exprs", perplexity=5, colour_by="total_features",size_by="total_counts",return_SCESet=FALSE) + ggtitle("tSNE plot")
```
Plot a summary of the QC
```{r create QC plots} 

plot.qc <- plot_grid(p1,p2,p3,p4,p5,p6, ncol = 2)

save_plot("results/QC_data/qc_summary_trapnell.png", plot.qc,
          ncol = 2, # we're saving a grid plot of 2 columns
          nrow = 2 # and 2 rows
          )


```

```{r plot tSNE} 
set.seed(1234)
out5 <- plotTSNE(sceset.red, exprs_values="exprs", perplexity=5, colour_by="total_features") + ggtitle("Trapnell 2014")
out5 <- plotTSNE(sceset.red, exprs_values="exprs", perplexity=5, colour_by="total_features") + ggtitle("Perplexity = 5")
out10 <- plotTSNE(sceset.red, exprs_values="exprs", perplexity=10, colour_by="total_features") + ggtitle("Perplexity = 10")
out20 <- plotTSNE(sceset.red, exprs_values="exprs", perplexity=20, colour_by="total_features") + ggtitle("Perplexity = 20") 
multiplot(out5, out10, out20, cols=3)
ggsave("results/QC_data/tsnelibsize_trapnell2014.pdf")
dev.off()
```

```{r remove  batch effects}

require(limma)

adj.exprs <- exprs(sceset.red)

adj.exprs <- removeBatchEffect(adj.exprs, batch=sceset.red$total_features )

norm_exprs(sceset.red) <- adj.exprs


plotTSNE(sceset.red, exprs_values="norm_exprs", perplexity=20, colour_by="phenoid",return_SCESet=FALSE) + ggtitle("adjusted for total features") 
plotTSNE(sceset.red, exprs_values="exprs", perplexity=20, colour_by="phenoid",return_SCESet=FALSE) + ggtitle("non adjusted") 


```
