---
title: "Import_QC_pbmsc"
output:
  pdf_document: default
  html_document: default
---
Note: decide if normalize as well bz ERCCC !!! 


Script and code follows "A step-by-step workflow for low-level analysis of single-cell RNA-seq data with Bioconductor". Aaron T. L. Lun1, Davis J. McCarthy2 and John C. Marioni. 

# Load and process data set, mixture of cells, Zheng et al. 2017. For the analysis the filtered cell matrices are used. However the filtering step is applied again to check dataset. Different cells are sampled and then the quality control is applied. Another option is to first apply  quality control and then mix cells., 
```{r load libraries, include=FALSE}
pdf("results/QC_data/QC_plots_Zhengmix.pdf")

#knittr options 
suppressPackageStartupMessages(library(MultiAssayExperiment))
suppressPackageStartupMessages(library(scater))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(plyr))
suppressPackageStartupMessages(library(scran))
suppressPackageStartupMessages(library(cowplot))


```


## load CD cells from 10xGenomics (https://support.10xgenomics.com/single-cell-gene-expression/datasets).

```{r load dataset}

# load multiassay dataset from 10xgenomjcs. https://support.10xgenomics.com/single-cell-gene-expression/datasets conquer
# extract counts, expand sparse matrix
b.cells <- read10XResults("/Users/angeloduo/Desktop/masterarbeit/scRNAseq_clustering_comparison/data/raw/b_cells/hg19") 
naive.cytotoxic <- read10XResults("/Users/angeloduo/Desktop/masterarbeit/scRNAseq_clustering_comparison/data/raw/naive_cytotoxic/hg19")
cd14.monocytes <- read10XResults("/Users/angeloduo/Desktop/masterarbeit/scRNAseq_clustering_comparison/data/raw/cd14_monocytes/hg19")
regulatory.t <- read10XResults("/Users/angeloduo/Desktop/masterarbeit/scRNAseq_clustering_comparison/data/raw/regulatory_t/hg19")

```
# sample from the original datasets
```{r load dataset}

# summary of data set
dim(assay(b.cells )) # 32738 x 2547
dim(assay(naive.cytotoxic )) # 32738  9143
dim(assay(cd14.monocytes )) # 32738   722
dim(assay(regulatory.t )) #  32738  7883

# rename cell labels in all datasets
colnames(b.cells) <-  array(paste0("b.cells", seq(1:ncol(b.cells))),dim =ncol(b.cells) )
colnames(naive.cytotoxic) <-  array(paste0("naive.cytotoxic", seq(1:ncol(naive.cytotoxic))),dim =ncol(naive.cytotoxic) )
colnames(cd14.monocytes ) <-  array(paste0("cd14.monocytes", seq(1:ncol(cd14.monocytes))),dim =ncol(cd14.monocytes) )
colnames(regulatory.t ) <-  array(paste0("regulatory.t", seq(1:ncol(regulatory.t))),dim =ncol(regulatory.t) )

### add phenoid
colData(b.cells)$phenoid <- array("b.cells", dim = ncol(b.cells) )
colData(naive.cytotoxic)$phenoid <- array("naive.cytotoxic", dim = ncol(naive.cytotoxic) )
colData(cd14.monocytes)$phenoid <- array("cd14.monocytes", dim = ncol(cd14.monocytes) )
colData(regulatory.t)$phenoid <- array("regulatory.t", dim = ncol(regulatory.t) )

## sample
b.cells <- b.cells[,sample( colnames(b.cells) , 500, replace = FALSE, prob = NULL) ]
naive.cytotoxic<- naive.cytotoxic[,sample( colnames(naive.cytotoxic) , 500, replace = FALSE, prob = NULL) ]
cd14.monocytes<- cd14.monocytes[,sample( colnames(cd14.monocytes) , 500, replace = FALSE, prob = NULL) ]
regulatory.t <- regulatory.t[,sample( colnames(regulatory.t ) , 500, replace = FALSE, prob = NULL) ]

```

# create an SCEset
The expr slot is populated with the count data; obtained from the length-scalaed , count-scaled tpms. The values are automatically log2 transformed. gene-level tpms for the tpm slot.
```{r create singlecellexperiment set}
# merge the stuff
sceset <-cbind(b.cells, naive.cytotoxic, cd14.monocytes,regulatory.t)

# change sparse matrix representation to expanded one 
counts(sceset) <-  counts(sceset) %>% as.matrix 
```
throw away features which are not expressed
```{r reduce expression matrix, throwing away features that are not expressed. noninformative features}
keep_features <- rowSums(counts(sceset) > 0) >= 1
table(keep_features)
sceset<- sceset[keep_features, ]
dim(sceset)
```
log transform
```{r log transform}
logcounts(sceset) <- log2(calculateCPM(sceset, 
                                            use.size.factors = FALSE) + 1)
```
save the raw data
```{r save raw data}
save( sceset,file = "data/sceset_raw_zhengmix.rda" )
```

check the names of the potential spike inns. Based on webresearch these are not spike inns but mitochondrial genes....
```{r check spike inns}

rowData(b.cells)$symbol[grep("^ERCC", rowData(sceset)$symbol ) ]
rowData(naive.cytotoxic)$symbol[grep("^ERCC", rowData(sceset)$symbol ) ]
rowData(cd14.monocytes)$symbol[grep("^ERCC", rowData(sceset)$symbol ) ]
rowData(regulatory.t)$symbol[grep("^ERCC", rowData(sceset)$symbol ) ]

```

# Identify the ERCCC spike-ins and mitochondrial data
There are no ERCCC spike-inn genes. 13 mitochondrial genes. ERCC are not really spike inns, UMI data contains no spike inns
```{r ERCC and mt genes }

is.spike <- grepl("^ERCC", rowData(sceset)$symbol )
is.mito <-  grepl("^MT-", rowData(sceset)$symbol ) 

rowData(sceset)$symbol[grep("^ERCC", rowData(sceset)$symbol ) ]
rowData(sceset)$symbol[grep("^MT-", rowData(sceset)$symbol ) ]


print( table(is.spike) )
print( table(is.mito) )
```


# Compute QC metrics, set mitochondrial features. 

```{r QC}
#isSpike(sceset) <- is.spike
sceset <- calculateQCMetrics(sceset )
colnames(colData(sceset))


```

# Overview of dataset

```{r overview}
par(mfrow=c(2,2))
hist(sceset$total_counts/1e3, xlab="Library sizes (thousands)", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
hist(sceset$total_features, xlab="Number of expressed genes", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
#hist(sceset$pct_counts_feature_controls_Mt, xlab="Mitochondrial proportion (%)", 
#    ylab="Number of cells", breaks=20, main="", col="grey80")
```
# Quality control on cells
The number of expressed genes shows a bimodal distribution. Use log transforamtion for a shift in lower libsizes. 
```{r  histogramms }
size.med <- median(log10(sceset$total_counts), na.rm = TRUE)
size.mad <- mad(log10(sceset$total_counts), center = size.med, na.rm = TRUE)
lower.limit.size <- 10^(size.med - 3 * size.mad)
    
feature.med <- median(log10(sceset$total_features), na.rm = TRUE)
feature.mad <- mad(log10(sceset$total_features), center =feature.med, na.rm = TRUE)
lower.limit.feat <- 10^(feature.med - 3 * feature.mad  )


df.hist <- as.data.frame(cbind("total_counts"=sceset$total_counts,"total_features"=sceset$total_features) )

hist.1 <- ggplot(df.hist, aes(x=total_counts))+
  geom_histogram()+
  labs(x="Library sizes ",y="Number of cells" )+
  geom_vline(xintercept=lower.limit.size, color=2)

hist.2 <- ggplot(df.hist, aes(x=total_features))+
  geom_histogram()+
  labs(x="Number of expressed genes",y="Number of cells" )+
  geom_vline(xintercept=lower.limit.feat, color=2)

```


# Filter dataset gene wise and by librarysize 
Remove cells with log-library sizes that are more than 3 median absolute deviations (MADs) below the median log-library size. 21 cells were dropped according to this criteria, regarding the untransformed count data. add plot with threshold...
```{r save unfiltered data }

libsize.drop <- isOutlier(sceset$total_counts, nmads=3, type="lower", log= TRUE)
feature.drop <- isOutlier(sceset$total_features, nmads=3, type="lower", log= TRUE)
#mito.drop <- isOutlier( sceset$pct_counts_feature_controls, nmads=3, type="higher" )

table( libsize.drop, feature.drop )

```
# Quality control using PCA and highest expressed genes
PCA plot based the quality metrics for each cell, e.g., the total number of reads, the total number of features and the proportion of mitochondrial or spike-in reads. 
```{r  Quality control using PCA }
p1 <- scater::plotPCA(sceset, exprs_values="logcounts",pca_data_input="pdata",colour_by="phenoid")
p2 <- scater::plotQC(sceset, type = "find-pcs", variable = "total_features")

```

Total 24 cells were filtered out.  
```{r overview filtered cells }

sceset.red <- sceset[,!(libsize.drop | feature.drop)]

data.frame(ByLibSize=sum(libsize.drop), ByFeature=sum(feature.drop),
    ByMito=sum(mito.drop), Remaining=ncol(sceset.red))

```


Plotting the 50 highest expressed genes. No ERCC spike-inn in the top 50. The profile should look relatively "flat".
```{r  highest expressed genes}
p3 <- plotQC(sceset.red, type = "highest-expression", n=50)

```

```{r mean freq expression}
plotQC(sceset.red, type = "exprs-freq-vs-mean")
```

```{r mean freq expression}
ave.counts <- rowMeans(counts(sceset.red))

ave.counts<- as.data.frame(log10(ave.counts))

p4 <- ggplot(ave.counts, aes(ave.counts))+geom_histogram(binwidth = 0.5)+geom_vline(aes(xintercept=log10(-3), color=1))+guides(color = "none")+labs( x = expression(Log[10]~"average count"), y = "frequency")
```


filter genes that were expressed only in one cell. Following Hemberg et al.
```{r  gene filtering part 2, eval=TRUE }
feature2drop <- apply(counts(sceset.red), 1, 
                      function(x) length(x[x > 1]) >= 2)
table(feature2drop )
scset.red <- sceset.red[feature2drop,]
```

Normalize data according to Lun et al. 2016 using pooled counts to estimate size factors. The size factors are positevely correlated with the library sizes for all cells. To avoid negative size factors, coming from to much zero counts per gene , a threshold is set to compute factors. Other possibility is to use a more stringet filtering in the previous step. 
Data is also normalized usong the mitochondrial genes, this step has probably to be deleted
```{r normalize through pool-based size factors }
hist(calcAverage(sceset), breaks = 10000, xlim = c(0,0.5))
abline(v=0.1)
high.ave <- calcAverage(sceset) >= 0.001



sceset<- computeSumFactors(sceset, subset.row=high.ave)
summary(sizeFactors(sceset))
plot(sizeFactors(sceset), sceset$total_counts, log="xy",
    ylab="Library size ", xlab="Size factor")
sceset <- normaliseExprs( sceset )


high.ave <- calcAverage(sceset.red) >= 0.001
sceset.red <- computeSumFactors(sceset.red,subset.row=high.ave)
summary(sizeFactors(sceset.red))
plot(sizeFactors(sceset.red), sceset.red$total_counts, log="xy",
    ylab="Library size ", xlab="Size factor")
sceset.red <- normaliseExprs( sceset.red )
assays(sceset.red)
```

# Save the normalized and cell filtered datasets
```{r save data }
# save unfiltered data
res <- sceset
save(res,file = "data/sceset_org_zhengmix.rda")

# save iltered data
res <- sceset.red
save(res,file = "data/sceset_red_zhengmix.rda")
```
Plot the proportion of explained variances
```{r explained variance} 
expl_vars <- c( "phenoid","log10_total_counts", "log10_total_features", "pct_dropout",
               "pct_counts_top_200_features", "log10_counts_feature_controls",
               "pct_counts_feature_controls")
p5 <- plotQC(scset.red, type = "explanatory-variables", variables = expl_vars)
```
plot tSNE represenatation
```{r tSNE}
p6 <- scater::plotTSNE(scset.red, exprs_values="logcounts", perplexity=5, colour_by="total_features",size_by="total_counts",return_SCESet=FALSE) + ggtitle("tSNE plot")
```

```{r create QC plots} 


plot.qc <- plot_grid(hist.1, hist.2,p3,p4,p5,p6, ncol = 2, labels="auto")

save_plot("results/QC_data/qc_summary_zheng.png", plot.qc,
          ncol = 2, # we're saving a grid plot of 2 columns
          nrow = 2 # and 2 rows
          )
dev.off()
```
