---
title: "Import_QC_pbmsc"
output:
  pdf_document: default
  html_document: default
---
Note: decide if normalize as well bz ERCCC !!! 


Script and code follows "A step-by-step workflow for low-level analysis of single-cell RNA-seq data with Bioconductor". Aaron T. L. Lun1, Davis J. McCarthy2 and John C. Marioni. 

# Load and process data set, mixture of cells, Zheng et al. 2017. For the analysis the filtered cell matrices are used. However the filtering step is applied again to check dataset. Different cells are sampled and then the quality control is applied. Another option is to first apply  quality control and then mix cells., 
```{r load libraries, include=FALSE}
pdf("results/QC_data/QC_plots_Zhengmix.pdf")

#knittr options 
suppressPackageStartupMessages(library(MultiAssayExperiment))
suppressPackageStartupMessages(library(scater))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(plyr))
suppressPackageStartupMessages(library(scran))
suppressPackageStartupMessages(library(cowplot))


```


## load CD cells from 10xGenomics (https://support.10xgenomics.com/single-cell-gene-expression/datasets).

```{r load dataset}
# load multiassay dataset from 10xgenomjcs. https://support.10xgenomics.com/single-cell-gene-expression/datasets conquer
b.cells <- read10XResults("/Users/angeloduo/Desktop/masterarbeit/scRNAseq_clustering_comparison/data/raw/b_cells/hg19", expand = TRUE)
naive.cytotoxic <- read10XResults("/Users/angeloduo/Desktop/masterarbeit/scRNAseq_clustering_comparison/data/raw/naive_cytotoxic/hg19", expand = TRUE)
cd14.monocytes <- read10XResults("/Users/angeloduo/Desktop/masterarbeit/scRNAseq_clustering_comparison/data/raw/cd14_monocytes/hg19", expand = TRUE)
regulatory.t <- read10XResults("/Users/angeloduo/Desktop/masterarbeit/scRNAseq_clustering_comparison/data/raw/regulatory_t/hg19", expand = TRUE)

#
par(mfrow=c(2,2))
b <- calculateQCMetrics(naive.cytotoxic )
hist(b$total_counts/1e3, xlab="Library sizes (thousands)", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
hist(b$total_features, xlab="Number of expressed genes", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
#
#
th <- calculateQCMetrics(cd4.t.helper)
hist(th$total_counts/1e3, xlab="Library sizes (thousands)", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
hist(th$total_features, xlab="Number of expressed genes", main="", 
    breaks=20, col="grey80", ylab="Number of cells")

#
# summary of data set
dim(assay(b.cells )) # 32738 x 2547
dim(assay(naive.cytotoxic )) # 32738  9143
dim(assay(cd14.monocytes )) # 32738   722
dim(assay(regulatory.t )) #  32738  7883
# rename cell labels in all datasets

colnames(b.cells) <-  array(paste0("b.cells", seq(1:ncol(b.cells))),dim =ncol(b.cells) )
colnames(naive.cytotoxic) <-  array(paste0("naive.cytotoxic", seq(1:ncol(naive.cytotoxic))),dim =ncol(naive.cytotoxic) )
colnames(cd14.monocytes ) <-  array(paste0("cd14.monocytes", seq(1:ncol(cd14.monocytes))),dim =ncol(cd14.monocytes) )
colnames(regulatory.t ) <-  array(paste0("regulatory.t", seq(1:ncol(regulatory.t))),dim =ncol(regulatory.t) )


### add phenoid
pData(b.cells)$phenoid <-array("b.cells", dim = ncol(b.cells) )
pData(naive.cytotoxic)$phenoid <-array("naive.cytotoxic", dim = ncol(naive.cytotoxic) )
pData(cd14.monocytes)$phenoid <-array("cd14.monocytes", dim = ncol(cd14.monocytes) )
pData(regulatory.t)$phenoid <-array("regulatory.t", dim = ncol(regulatory.t) )


## sample
b.cells <- b.cells[,sample( colnames(b.cells) , 500, replace = FALSE, prob = NULL) ]
naive.cytotoxic<- naive.cytotoxic[,sample( colnames(naive.cytotoxic) , 500, replace = FALSE, prob = NULL) ]
cd14.monocytes<- cd14.monocytes[,sample( colnames(cd14.monocytes) , 500, replace = FALSE, prob = NULL) ]
regulatory.t <- regulatory.t[,sample( colnames(regulatory.t ) , 500, replace = FALSE, prob = NULL) ]


scset <- mergeSCESet(b.cells, naive.cytotoxic)
scset <- mergeSCESet(scset, cd14.monocytes)
scset <- mergeSCESet(scset, regulatory.t)

```


```{r check spike inns}
b.cells@featureData$symbol[grep("^ERCC", scset@featureData$symbol )]
naive.cytotoxic@featureData$symbol[grep("^ERCC", scset@featureData$symbol )]
cd14.monocytes@featureData$symbol[grep("^ERCC", scset@featureData$symbol )]
regulatory.t@featureData$symbol[grep("^ERCC", scset@featureData$symbol )]

```
# reduce expression set
Remove genes that are not expressed in any cell
Keep only those features that are observed in at least 1 cell. 435 features are filtered out. The original counts are used.

```{r reduce}
keep_features <- rowSums(counts(scset) > 0) >= 1
table(keep_features)
scset <-scset[keep_features, ]
dim(scset)
```




# Identify the ERCCC spike-ins and mitochondrial data
There are no ERCCC spike-inn genes. 13 mitochondrial genes. ERCC are not really spike inns, UMI data contains no spike inns
```{r ERCC and mt genes }
is.spike <- grepl("^ERCC", scset@featureData$symbol )
is.mito <-  grepl("^MT-", scset@featureData$symbol ) 


scset@featureData$symbol[grep("^ERCC", scset@featureData$symbol )]
scset@featureData$symbol[grep("^MT-", scset@featureData$symbol ) ]


print( table(is.spike) )
print( table(is.mito) )
```





# Compute some QC metrics, set mitochondrial features. 

```{r QC}

scset <- calculateQCMetrics(scset, feature_controls = list(Mt=grepl("^MT-",  scset@featureData$symbol) ))
setSpike(scset) <- "Mt"

hist(scset$pct_counts_feature_controls  , xlab="Mitochondrial proportion (%)", 
    ylab="Number of cells", main="", col="grey80")


```

# Overview of dataset

```{r overview}
par(mfrow=c(2,2))
hist(scset$total_counts/1e3, xlab="Library sizes (thousands)", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
hist(scset$total_features, xlab="Number of expressed genes", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
hist(scset$pct_counts_feature_controls_Mt, xlab="Mitochondrial proportion (%)", 
    ylab="Number of cells", breaks=20, main="", col="grey80")
```
# Quality control on cells
The number of expressed genes shows a bimodal distribution. Use log transforamtion for a shift in lower libsizes. 
```{r  histogramms }
size.med <- median(log10(scset$total_counts), na.rm = TRUE)
size.mad <- mad(log10(scset$total_counts), center = size.med, na.rm = TRUE)
lower.limit.size <- 10^(size.med - 3 * size.mad)
    
feature.med <- median(log10(scset$total_features), na.rm = TRUE)
feature.mad <- mad(log10(scset$total_features), center = size.med, na.rm = TRUE)
lower.limit.feat <- 10^(feature.med - 3 * feature.mad  )

mito.med <- median((scset$pct_counts_feature_controls_Mt), na.rm = TRUE)
mito.mad <- mad((scset$pct_counts_feature_controls_Mt), center = size.med, na.rm = TRUE)
upper.limit.mito <- (mito.med + 3 * mito.mad  )



par(mfrow=c(2,2))
hist(scset$total_counts, xlab="Library sizes ", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
abline(v=lower.limit.size,col=2, lty=2 )
hist(scset$total_features, xlab="Number of expressed genes", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
abline(v=lower.limit.feat, col=2, lty=2 )
hist(scset$pct_counts_feature_controls_Mt, xlab="Mitochondrial proportion (%)", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
abline(v=upper.limit.mito, col=2, lty=2 )

```


# Filter dataset gene wise and by librarysize 
Remove cells with log-library sizes that are more than 3 median absolute deviations (MADs) below the median log-library size. 21 cells were dropped according to this criteria, regarding the untransformed count data. add plot with threshold...
```{r save unfiltered data }

libsize.drop <- isOutlier(scset$total_counts, nmads=3, type="lower", log= TRUE)
feature.drop <- isOutlier(scset$total_features, nmads=3, type="lower", log= TRUE)
mito.drop <- isOutlier( scset$pct_counts_feature_controls, nmads=3, type="higher" )

table( libsize.drop, feature.drop, mito.drop )

```
# Quality control using PCA and highest expressed genes
PCA plot based the quality metrics for each cell, e.g., the total number of reads, the total number of features and the proportion of mitochondrial or spike-in reads. 
```{r  Quality control using PCA }
p1 <- scater::plotPCA(scset, exprs_values="exprs",pca_data_input="pdata",colour_by="phenoid")
p2 <- scater::plotQC(scset, type = "find-pcs", variable = "total_features")


```

Total 24 cells were filtered out.  
```{r overview filtered cells }

scset.red <- scset[,!(libsize.drop | feature.drop |  mito.drop)]

data.frame(ByLibSize=sum(libsize.drop), ByFeature=sum(feature.drop),
    ByMito=sum(mito.drop), Remaining=ncol(scset.red))

```


Plotting the 50 highest expressed genes. No ERCC spike-inn in the top 50. The profile should look relatively "flat".
```{r  highest expressed genes}
p3 <- plotQC(scset.red, type = "highest-expression", n=50)

```

```{r mean freq expression}
plotQC(sceset.red, type = "exprs-freq-vs-mean")
```

```{r mean freq expression}
ave.counts <- rowMeans(counts(scset.red))

ave.counts<- as.data.frame(log10(ave.counts))

p4 <- ggplot(ave.counts, aes(ave.counts))+geom_histogram(binwidth = 0.5)+geom_vline(aes(xintercept=log10(1), color=1))+guides(color = "none")+labs( x = expression(Log[10]~"average count"), y = "frequency")
```


filter genes that were expressed only in one cell. Following Hemberg et al.
```{r  gene filtering part 2, eval=TRUE }
feature2.drop <- apply(counts(scset.red), 1, 
                      function(x) length(x[x > 1]) >= 2)
table(feature2.drop )
scset.red <- scset.red[feature2.drop,]
```

Normalize data according to Lun et al. 2016 using pooled counts to estimate size factors. The size factors are positevely correlated with the library sizes for all cells. To avoid negative size factors, coming from to much zero counts per gene , a threshold is set to compute factors. Other possibility is to use a more stringet filtering in the previous step. 
Data is also normalized usong the mitochondrial genes, this step has probably to be deleted
```{r normalize through pool-based size factors }
high.ave <- calcAverage(scset.red) >= 0.1
scset.red <- computeSumFactors(scset.red,subset.row=high.ave)
summary(sizeFactors(scset.red))
plot(sizeFactors(scset.red), scset.red$total_counts, log="xy",
    ylab="Library size ", xlab="Size factor")
scset.red <- normalize( scset.red )

```

# Save the normalized and cell filtered datasets
```{r save data }
res <- scset.red
save(res,file = "data/sceset_red_zhengmix.rda")

```
Plot the proportion of explained variances
```{r explained variance} 
expl_vars <- c( "phenoid","log10_total_counts", "log10_total_features", "pct_dropout",
               "pct_counts_top_200_features", "log10_counts_feature_controls",
               "pct_counts_feature_controls")
p5 <- plotQC(scset.red, type = "explanatory-variables", variables = expl_vars)
```
plot tSNE represenatation
```{r tSNE}
p6 <- plotTSNE(scset.red, exprs_values="exprs", perplexity=5, colour_by="total_features",size_by="total_counts",return_SCESet=FALSE) + ggtitle("tSNE plot")
```

```{r create QC plots} 


plot.qc <- plot_grid(p1,p2,p3,p4,p5,p6, ncol = 2)

save_plot("results/QC_data/qc_summary_zheng.png", plot.qc,
          ncol = 2, # we're saving a grid plot of 2 columns
          nrow = 2 # and 2 rows
          )
dev.off()
```
